File: iosApp/iosApp/Platform/SnackbarMessageHandler.swift

import Shared

class SnackbarMessageHandlerImpl : SnackBarMessageHandler {
    private let snackbarManager: SnackbarManager
    
    init(snackbarManager: SnackbarManager) {
        self.snackbarManager = snackbarManager
    }
    
    func showSnackBarMessage(message: SnackBarMessage, callBack: @escaping (SharedSnackbarResult) -> Void) {
        Task {
            let result = await snackbarManager.show(message.message, actionTitle: message.actionLabel, duration: message.duration.timeInterval)
            callBack(result == .dismissed ? SharedSnackbarResult.dismissed : SharedSnackbarResult.actionperformed)
        }
    }
}

File: shared/platform/src/commonMain/kotlin/me/andannn/aniflow/platform/SnackBarMessageHandler.kt
/*
 * Copyright 2025, the AniflowKMP project contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package me.andannn.aniflow.platform

interface SnackBarMessageHandler {
    fun showSnackBarMessage(
        message: SnackBarMessage,
        callBack: (SharedSnackbarResult) -> Unit = {},
    )
}

enum class SharedSnackbarResult {
    Dismissed,
    ActionPerformed,
}

enum class SnackbarShowDuration {
    Short,
    Long,
    Indefinite,
}

interface SnackBarMessage {
    val message: String
    val duration: SnackbarShowDuration
    val actionLabel: String?
    val withDismissAction: Boolean
}
